File: TelerikUIExamples/TelerikUIExamples/ChartDocsLineSeries.m
//
//  ChartDocsLineSeries.m
//  TelerikUIExamples
//
//  Copyright © 2016 Telerik. All rights reserved.
//

#import "ChartDocsLineSeries.h"
#import <TelerikUI/TelerikUI.h>

@implementation ChartDocsLineSeries
{
    NSMutableArray *expensesData;
    NSMutableArray *incomesData;
    NSMutableArray *profitData;
    TKChart *chart;
}

- (void)viewDidLoad
{
    [super viewDidLoad];
    // Do any additional setup after loading the view.
    expensesData = [[NSMutableArray alloc] init];
    incomesData = [[NSMutableArray alloc] init];
    profitData = [[NSMutableArray alloc] init];
   
    [self lineSeries];
    
    //chart
    [self visualAppearance];
}

- (void) lineSeries
{
    // >> chart-line
    NSArray *expensesValues = @[@60, @30, @50, @32, @31];
    NSArray *incomesValues = @[@70, @75, @58, @59, @88];
    NSArray *profitValues = @[@10, @45, @8, @27, @57];
    NSArray *categories = @[@"Greetings", @"Perfecto", @"NearBy", @"Family Store", @"Fresh & Green"];
    for (int i = 0; i < categories.count ; i++) {
        [expensesData addObject:[TKChartDataPoint dataPointWithX:categories[i] Y:expensesValues[i]]];
        [incomesData addObject:[TKChartDataPoint dataPointWithX:categories[i] Y:incomesValues[i]]];
        [profitData addObject:[TKChartDataPoint dataPointWithX:categories[i] Y:profitValues[i]]];
    }
    
    TKChartLineSeries* seriesForExpenses = [[TKChartLineSeries alloc] initWithItems:expensesData];
    seriesForExpenses.title = @"Expenses";
    [chart addSeries:seriesForExpenses];
    
    TKChartLineSeries* seriesForIncomes = [[TKChartLineSeries alloc] initWithItems:incomesData];
    seriesForIncomes.title = @"Incomes";
    [chart addSeries:seriesForIncomes];
    
    TKChartLineSeries* seriesForProfit = [[TKChartLineSeries alloc] initWithItems:profitData];
    seriesForProfit.title = @"Profit";
    [chart addSeries:seriesForProfit];
    // << chart-line
    
    chart.legend.hidden = NO;
}

- (void) input
{
    TKChartLineSeries* seriesForProfit = [[TKChartLineSeries alloc] initWithItems:profitData];
    seriesForProfit.selection = TKChartSeriesSelectionSeries;
    seriesForProfit.marginForHitDetection = 30.f;
    [chart addSeries:seriesForProfit];
}

- (void) visualAppearance
{
    // >> chart-line-series-stroke
    TKChartLineSeries *seriesForProfit = [[TKChartLineSeries alloc] initWithItems:profitData];
    seriesForProfit.style.palette = [[TKChartPalette alloc] init];
    TKChartPaletteItem *palleteItem = [[TKChartPaletteItem alloc] init];
    palleteItem.stroke = [TKStroke strokeWithColor:[UIColor greenColor]];
    [seriesForProfit.style.palette addPaletteItem:palleteItem];
    [chart addSeries:seriesForProfit];
    // << chart-line-series-stroke
}

@end

File: TelerikUIExamplesInSwift/TelerikUIExamplesInSwift/ChartDocsLineSeries.swift
//
//  ChartDocsLineSeries.swift
//  TelerikUIExamplesInSwift
//
//  Copyright © 2016 Telerik. All rights reserved.
//

class ChartDocsLineSeries: UIViewController {

    var chart = TKChart()
    var expensesData = [TKChartDataPoint]()
    var incomesData = [TKChartDataPoint]()
    var profitData = [TKChartDataPoint]()
    
    override func viewDidLoad() {
        super.viewDidLoad()
        chart.frame = CGRectInset(self.view.bounds, 10, 10)
        self.view.addSubview(chart)
        self.lineSeries()
    }
    
    func lineSeries() {
        // >> chart-line-swift
        let categories = ["Greetings", "Perfecto", "NearBy", "Family Store", "Fresh & Green" ]
        let expensesValues = [60, 30, 50, 32, 31]
        let incomesValues = [70, 75, 58, 59, 88]
        let profitValues = [10, 45, 8, 27, 57]
        
        for i in 0 ..< categories.count {
            expensesData.append(TKChartDataPoint(x: categories[i], y: expensesValues[i]))
            incomesData.append(TKChartDataPoint(x: categories[i], y: incomesValues[i]))
            profitData.append(TKChartDataPoint(x: categories[i], y: profitValues[i]))
        }
        
        let seriesForExpenses = TKChartLineSeries(items: expensesData)
        seriesForExpenses.title = "Expenses"
        chart.addSeries(seriesForExpenses)
        
        let seriesForIncomes = TKChartLineSeries(items: incomesData)
        seriesForIncomes.title = "Incomes"
        chart.addSeries(seriesForIncomes)
        
        let seriesForProfit = TKChartLineSeries(items: profitData)
        seriesForProfit.title = "Profit"
        chart.addSeries(seriesForProfit)
        chart.legend.hidden = false
        // << chart-line-swift
    }
    
    func selection() {
        let seriesForProfit = TKChartLineSeries(items: profitData)
        seriesForProfit.selection = TKChartSeriesSelection.Series
        seriesForProfit.marginForHitDetection = 30.0
        chart.addSeries(seriesForProfit)
    }
    
    func visualAppearance() {
        // >> chart-line-series-stroke-swift
        let seriesForProfit = TKChartLineSeries(items: profitData)
        seriesForProfit.style.palette = TKChartPalette()
        let paletteItem = TKChartPaletteItem()
        paletteItem.stroke = TKStroke(color: UIColor.greenColor())
        seriesForProfit.style.palette!.addPaletteItem(paletteItem)
        chart.addSeries(seriesForProfit)
        // << chart-line-series-stroke-swift
    }
}
